docs/source/python/json.rst

```diff
@@ -0,0 +1,114 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow.json
+.. _json:
+
+Reading JSON files
+==================
+
+Arrow supports reading columnar data from JSON files.  A JSON file consists
+of multiple JSON objects, one per line, representing individual data rows.
+For example, this file represents two rows of data with four columns "a", "b",
+"c", "d":
+
+.. code-block:: json
+
+   {"a": 1, "b": 2.0, "c": "foo", "d": false}
+   {"a": 4, "b": -5.5, "c": null, "d": true}
+
+The features currently offered are the following:
+
+* multi-threaded or single-threaded reading
+* automatic decompression of input files (based on the filename extension,
+  such as ``my_data.json.gz``)
+* sophisticated type inference (see below)
+
+
+Usage
+-----
+
+JSON reading functionality is available through the :mod:`pyarrow.json` module.
+In many cases, you will simply call the :func:`read_json` function
+with the file path you want to read from::
+
+   >>> from pyarrow import json
+   >>> fn = 'my_data.json'
+   >>> table = json.read_csv(fn)
+   >>> table
+   pyarrow.Table
+   a: int64
+   b: double
+   c: string
+   d: bool
+   >>> table.to_pandas()
+      a    b     c      d
+   0  1  2.0   foo  False
+   1  4 -5.5  None   True
+
+
+Automatic Type Inference
+------------------------
+
+Arrow :ref:`data types <data.types>` are inferred from the JSON types and
+values of each column:
+
+* JSON null values convert to the ``null`` type, but can fall back to any
```

Comment:
WIth "fall back", you mean that it can be "upcasted" (maybe also not the correct term) if a subsequent row has a different type? 
If my understanding is correct, "fall back" reads a bit strange. Maybe "get promoted"?

Reply:
Although, while now reading further, the "fall back" in the timestamp example is very logical .. so maybe it is a good term after all.

Reply:
"upcasted" and "promoted" are technical jargon that I find a bit confusing myself :-)